src/main/java/frc/robot/Subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,95 @@
+package frc.robot.Subsystems;
+
+
+import java.util.function.DoubleSupplier;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+/** A robot arm subsystem that moves with a motion profile. */
+public class ArmSubsystem extends SubsystemBase {
+  private CANSparkMax armMotor;
+  private SparkMaxPIDController m_pidController;
+  private RelativeEncoder m_encoder;
+  public double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput;
+
+  public ArmSubsystem() {
+    armMotor = new CANSparkMax(Constants.ARM_MOTOR_ID, MotorType.kBrushless);
+    m_encoder = armMotor.getEncoder();
+
+    armMotor.setIdleMode(IdleMode.kBrake);
+    armMotor.restoreFactoryDefaults();
+    m_pidController = armMotor.getPIDController();
+
+    // PID coefficients
+    kP = 0.1; 
+    kI = 1e-4;
+    kD = 1; 
+    kIz = 0; 
+    kFF = 0; 
+    kMaxOutput = 1; 
+    kMinOutput = -1;
+
+    // set PID coefficients
+    m_pidController.setP(kP);
+    m_pidController.setI(kI);
+    m_pidController.setD(kD);
+    m_pidController.setIZone(kIz);
+    m_pidController.setFF(kFF);
+    m_pidController.setOutputRange(kMinOutput, kMaxOutput);
+
+    // display PID coefficients on SmartDashboard
+    SmartDashboard.putNumber("P Gain", kP);
+    SmartDashboard.putNumber("I Gain", kI);
+    SmartDashboard.putNumber("D Gain", kD);
+    SmartDashboard.putNumber("I Zone", kIz);
+    SmartDashboard.putNumber("Feed Forward", kFF);
+    SmartDashboard.putNumber("Max Output", kMaxOutput);
+    SmartDashboard.putNumber("Min Output", kMinOutput);
+    SmartDashboard.putNumber("Set Rotations", 0);
+  }
+
+
+ 
+  public void moveArm(DoubleSupplier joystickPos) {
```

Comment:
So... this reads input from the joystick on a periodic basis and moves the arm by +/- 0.1rad each time this is called?

Reply:
when only kP is nonzero and set to 0.1 yes